public/js/grid/timetable_util.js

```diff
@@ -175,13 +174,13 @@ function saveCookies(courses, sections) {
     'use strict';
 
     if (courses !== undefined) {
-        setCookie("selected-courses", JSON.stringify(courses).replace(/,/g, '_').replace(/[\"\[\]]/g, ''));
+        setCookie("selected-courses", courses.join('_').replace(/[\"\[\]]/g, ''));
```

Comment:
It doesn't look like it, but when trying it out I've come across what appears to be an unrelated bug...